Code/Framework/AzToolsFramework/AzToolsFramework/Archive/ArchiveComponent.cpp

```diff
@@ -141,320 +120,474 @@ namespace AzToolsFramework
             }
 
-            AzFramework::ProcessLauncher::ProcessLaunchInfo info;
-            info.m_commandlineParameters = exePath + " " + commandLineArgs;
-            
-            info.m_showWindow = false;
-            if (!workingDir.empty())
-            {
-                info.m_workingDirectory = workingDir;
-            }
-            AZStd::unique_ptr<AzFramework::ProcessWatcher> watcher(AzFramework::ProcessWatcher::LaunchProcess(info, AzFramework::ProcessCommunicationType::COMMUNICATOR_TYPE_STDINOUT));
+            bool success = true; // starts true and turns false when any error is encountered.
+            AZ::IO::Path basePath{ workingDirectory };
 
-            AZStd::string consoleOutput;
-            AZ::u32 exitCode = static_cast<AZ::u32>(SevenZipExitCode::UserStoppedProcess);
-            if (watcher)
+            auto PerLineCallback = [&success, &basePath, &archive](AZStd::string_view filePathLine) -> void
             {
-                // callback requires output captured from 7z 
-                if (captureOutput)
+                AZStd::vector<char> fileBuffer;
+                AZStd::string fullPath = (basePath / filePathLine).c_str();
+                if (ArchiveUtils::ReadFile(fullPath, AZ::IO::OpenMode::ModeRead, fileBuffer))
                 {
-                    AZStd::string consoleBuffer;
-                    while (watcher->IsProcessRunning(&exitCode))
-                    {
-                        if (!taskHandle.IsNull())
-                        {
-                            AZStd::unique_lock<AZStd::mutex> lock(m_threadControlMutex);
-                            if (m_threadInfoMap[taskHandle].shouldStop)
-                            {
-                                watcher->TerminateProcess(static_cast<AZ::u32>(SevenZipExitCode::UserStoppedProcess));
-                            }
-                        }
-                        watcher->WaitForProcessToExit(g_sleepDuration, &exitCode);
-                        AZ::u32 outputSize = watcher->GetCommunicator()->PeekOutput();
-                        if (outputSize)
-                        {
-                            consoleBuffer.resize(outputSize);
-                            watcher->GetCommunicator()->ReadOutput(consoleBuffer.data(), outputSize);
-                            consoleOutput += consoleBuffer;
-                        }
-                    }
+                    int result = archive->UpdateFile(
+                        filePathLine, fileBuffer.data(), fileBuffer.size(), s_compressionMethod,
+                        s_compressionLevel, s_compressionCodec);
+
+                    bool thisSuccess = (result == AZ::IO::ZipDir::ZD_ERROR_SUCCESS);
+                    success = (success && thisSuccess);
+                    AZ_Error(
+                        s_traceName, thisSuccess, "Error %d encountered while adding '%.*s' to archive '%s'", result,
+                        AZ_STRING_ARG(filePathLine), archive->GetFullPath());
```

Comment:
The `archive->GetFullPath()` function returns an AZ::IO::PathView.
The `.Native()` function is needed to return a string_view

```suggestion
                        s_traceName, thisSuccess, "Error %d encountered while adding '%.*s' to archive '%.*s'", result,
                        AZ_STRING_ARG(filePathLine), AZ_STRING_ARG(archive->GetFullPath().Native()));
```